os/kernel/task_monitor/task_monitor.c

```diff
@@ -55,26 +132,36 @@ int task_monitor(int argc, char *argv[])
 		/* Check if registered tasks/pthread is alive or not.
 		 * If one of them is not alive, task monitor resets the system.
 		 */
-		for (pid_idx = 0; pid_idx < CONFIG_MAX_TASKS; pid_idx++) {
-			if (monitored_tasks_list[pid_idx] == MONITORED) {
-				tcb = sched_gettcb(pid_idx);
-				if (tcb == NULL) {
-					/* Invalid operation */
-					boardctl(BOARDIOC_RESET, 0);
-				}
+		for (time_idx = 0; time_idx < TASK_MONITOR_CHECK_TIME; ++time_idx) {
+
+			if (g_que_list[time_idx].remaining_count == 0) {
+
+				next_mon = (task_monitor_node_t *)dq_peek(&g_que_list[time_idx - 1].q);
+
+				while (next_mon != NULL) {
+
+					tcb = sched_gettcb(next_mon->pid);
 
-				if (tcb->is_active == false) {
-					/* There is not alive task/pthread.
-					 * System will be reset.
-					 */
-					boardctl(BOARDIOC_RESET, 0);
-				} else {
-					/* Reset the registered task's/pthread's active flag. */
-					tcb->is_active = false;
+					if (tcb == NULL && tcb->is_active == false) {
```

Comment:
I think `tcb == NULL || tcb->is_active == false` is correct.
Please check this.